### PR TITLE
Added support for in-place update for replication.userManaged.replicas.customerManagedEncryption to google_secret_manager_secret resource

### PR DESCRIPTION
**Pull request by abheda-crest**
Added support for the in-place update of the `replication.userManaged.replicas.customerManagedEncryption` to the `google_secret_manager_secret` resource.
fixes https://github.com/hashicorp/terraform-provider-google/issues/15011

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
secretmanager: supported in-place update for `replication.user_managed.replicas.customer_managed_encryption` field in `google_secret_manager_secret` resource
```

### CODE REVIEW

**Comment by shuyama1** [comment]
@abheda-crest Thanks for making the change. Quick question: is replication.user_managed.replicas.customer_managed_encryption a mutable subfield of a chain of immutable fields?

**Comment by abheda-crest** [comment]
> @abheda-crest Thanks for making the change. Quick question: is replication.user_managed.replicas.customer_managed_encryption a mutable subfield of a chain of immutable fields?

Yeah. replication.user_managed and replication.user_managed.replicas are immutable fields. Patching the `replication.userManaged` field to `replication.automatic` throws an API error 400 `Existing secret has user-managed replication, but requested secret does not. Updating secret replication is not supported`. Patching the `replication.userManaged.replicas` field to add an extra replica throws an API error 400 `Adding or removing replicas from Secret user-managed replication is not supported. Number of replicas requested: [2], number of existing replicas: [1]`. Hence, we can say that replication.user_managed.replicas.customer_managed_encryption is a mutable subfield of a chain of immutable fields.

**Comment by shuyama1** [comment]
@abheda-crest Thank you for the ping. I'm taking a review now and will merge the PR if no changes needed.

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb [hunk 1]
```diff
@@ -426,3 +484,150 @@ resource "google_secret_manager_secret_version" "secret-version-4" {
...
+  
+  labels = {
+    label = "my-label"
+  }
+  replication {
+    user_managed {
+      replicas {
+        location = "us-central1"
+      }
+    replicas {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:519]:
```suggestion
      replicas {
```
nit-picks

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb [hunk 2]
```diff
@@ -426,3 +484,150 @@ resource "google_secret_manager_secret_version" "secret-version-4" {
...
+  }
+  replication {
+    user_managed {
+      replicas {
+        location = "us-central1"
+        customer_managed_encryption {
+          kms_key_name = "%{kms_key_name_central}"
+        }
+      }
+    replicas {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:567]:
```suggestion
      replicas {
```

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb [hunk 3]
```diff
@@ -426,3 +484,150 @@ resource "google_secret_manager_secret_version" "secret-version-4" {
...
+  }
+  replication {
+    user_managed {
+      replicas {
+        location = "us-central1"
+        customer_managed_encryption {
+          kms_key_name = "%{kms_key_name_central_other}"
+        }
+      }
+    replicas {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb:618]:
```suggestion
      replicas {
```

**Review by shuyama1 — COMMENTED** [review]
LGTM, only some nit-picks. Thanks for adding thorough testing! I've checked the debug logs and confirmed that the resource was update in-place instead of recreated.

File: mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb (hunk 1 above)
Comment (abheda-crest) [reply mmv1/third_party/terraform/services/secretmanager/resource_secret_manager_secret_test.go.erb]: Thanks for the review! I've added the extra whitespaces.

**Review by shuyama1 — APPROVED** [review]
Thank you!

**Pull request merged**